### PR TITLE
fix(types): add vitest support for AsymmetricMatchers

### PR DESCRIPTION
**Pull request by rluvaton**
Fix https://github.com/jest-community/jest-extended/issues/548

@SimenB should I update the package version?

### Housekeeping

- [ ] Unit tests
- [ ] Documentation is up to date
- [ ] No additional lint warnings
- [x] [Typescript definitions](https://github.com/jest-community/jest-extended/blob/main/types/index.d.ts) are added/updated where relevant

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -876,3 +879,8 @@ declare namespace jest {
   // eslint-disable-next-line @typescript-eslint/no-empty-interface
   interface InverseAsymmetricMatchers extends Expect {}
 }
+
+declare namespace Vi {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface AsymmetricMatchersContaining extends CustomMatchers {}
+}
```
Comment (rluvaton) [inline types/index.d.ts:886]: Took the idea from

https://github.com/vitest-dev/vitest/blob/199776767a10bcd9f241e36a5cdd578c7d7c0dc2/test/core/test/jest-expect.test.ts#L9-L17

File: types/index.d.ts [hunk 2]
```diff
@@ -1,5 +1,431 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+interface CustomMatchers {
+  /**
+   * Note: Currently unimplemented
+   * Passing assertion
+   *
+   * @param {String} message
+   */
+  pass(message: string): void;
```
Comment (SimenB) [inline types/index.d.ts:10]: why are these changed from `R` to `void`?

**Comment by SimenB** [comment]
>@SimenB should I update the package version?

No need, I'll do that when publishing

File: types/index.d.ts (hunk 2 above)
Comment (rluvaton) [reply types/index.d.ts:10]: Weird, I just copy pasted

Comment (rluvaton) [reply types/index.d.ts:10]: I'll do it again in a few minutes

Comment (rluvaton) [reply types/index.d.ts:10]: Never mind, there shouldn't be any `R` I copy pasted from the Expect interface
